addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -20,6 +20,44 @@
 class PaymentTransaction(models.Model):
     _inherit = 'payment.transaction'
 
+    @api.model
+    def _validate_and_sanatize_phone_number(self, phone):
+        """ Sanatize and validate phone number if phone number is valid and if not
+        valid then raise validation error.
+        :param char phone
+        :return: Sanatized and validated phone number.
+        :rtype: char
+        """
+        phone = self.partner_phone
+        error_message = _("The phone number is missing.")
+        if phone:
+            # sanitize partner phone
+            country_code = self.partner_country_id.code
+            country_phone_code = self.partner_country_id.phone_code
+            phone_info = phone_sanitize_numbers([phone], country_code, country_phone_code)
+            phone = phone_info[self.partner_phone]['sanitized']
+            error_message = phone_info[self.partner_phone]['msg']
+        if not phone:
+            raise ValidationError(_("Error during the phone validation for Razorpay usage. The provided error is %s", error_message))
+        return phone
```

Comment:
`phone_sanitize_numbers` support empty string, so you could:

```suggestion
        phone = self.partner_phone or ''
        country_code = self.partner_country_id.code
        country_phone_code = self.partner_country_id.phone_code
        phone_info = phone_sanitize_numbers([phone], country_code, country_phone_code)[phone]
        phone, error_message = phone_info['sanitized'], phone_info['msg']
        if not phone:
            error_message = error_message or _("The phone number is missing.")
            raise ValidationError(_(
                "Error during the phone validation for Razorpay usage. The provided error is %s",
                error_message,
            ))
        return phone
```